src/Storages/StorageDistributed.cpp

```diff
@@ -800,6 +800,9 @@ void StorageDistributed::checkAlterIsPossible(const AlterCommands & commands, Co
                     ErrorCodes::ALTER_OF_COLUMN_IS_FORBIDDEN);
             }
         }
+
+        if (command.ttl)
+            throw Exception("Engine Distributed doesn't support TTL clause", ErrorCodes::NOT_IMPLEMENTED);
```

Comment:
The same applies for other tables (i.e. Buffer/Merge/...), so maybe it is better to look at supports_ttl storage feature.
And also worth adding a test to avoid problems in future (`ALTER`/`DETACH`/`ATTACH` should be enough I guess)

Reply:
OK, I checked it, and currently only `merge tree` supports ttl, so I want to add the `supportsTTL` interface to `IStorage`. What do you think?